src/lib/resource.js

```diff
@@ -1,51 +1,139 @@
-  } catch (error) {
-    auditLogger.error(`Resource Queue: Unable to retrieve title for Resource (ID: ${resourceId} URL: ${resourceUrl}), please make sure this is a valid address:`, error);
-    throw Error(error); // We must rethrow the error here to ensure the job is retried.
-  }
-};
-
-export {
-  axios,
-  getResourceMetaDataJob,
-};
+/* eslint-disable import/prefer-default-export */
+import httpCodes from 'http-codes';
+import axios from 'axios';
+import { auditLogger, logger } from '../logger';
+import { Resource } from '../models';
+
+const getPageScrapeValues = async (resourceUrl, isEclkc) => {
+  // Use axios to get url info.
+  const res = await axios.get(resourceUrl, { maxRedirects: 3 });
+
+  // Scrape Title & National Centers from ECLKC resources.
+  const foundTitle = res.data.match(/<title[^>]*>([^<]+)<\/title>/);
+  // eslint-disable-next-line max-len
+  // Find the meta tag with name="national-centers".
+  const foundNationalCenters = isEclkc ? res.data.match(/<meta[^>]*=*"national-centers"[^>]*content="([^"]+)"[^>]*>/) : null;
+
+  // Update object to save.
+  const updateObj = {};
+
+  // If we found a title, add to update the resource.
+  if (foundTitle && foundTitle.length >= 1) {
+    // Get title.
+    const titleToUpdate = foundTitle[1].trim();
+
+    // Update URL in DB.
+    updateObj.title = titleToUpdate;
+  }
+
+  // If we found a national centers, add to update the resource.
+  if (foundNationalCenters && foundNationalCenters.length >= 1) {
+    // Get national centers.
+    const nationalCentersToUpdate = foundNationalCenters[1].trim();
+
+    // Update URL in DB.
+    updateObj.nationalCenters = nationalCentersToUpdate;
+  }
+
+  // If we have values to update then update the resource, else retry.
+  if (Object.keys(updateObj).length > 0) {
+    const updatedCnt = await Resource.update(updateObj, {
+      where: { url: resourceUrl },
+      individualHooks: false,
+    });
+
+    // If we don't update anything throw an error to retry.
+    if (updatedCnt[0] === 0) {
+      throw Error('Failed to update resource.');
+    }
+
+    return updatedCnt[0];
+  }
+
+  return 0;
+};
+
+const getMetadataValues = async (resourceId, resourceUrl, isEclkc) => {
+  // Attempt to get the resource metadata (if valid ECLKC resource).
+  // Sample: https://eclkc.ohs.acf.hhs.gov/mental-health/article/head-start-heals-campaign?_format=json
+  if (isEclkc) {
+    try {
+      const metadataUrl = `${resourceUrl}?_format=json`;
+      const res = await axios.get(metadataUrl, { maxRedirects: 3 });
+      const metadata = res.data;
+
+      // get created.
+      const { created } = metadata;
+      // get changed.
+      const { changed } = metadata;
+      // get title.
+      const { title } = metadata;
+      // get field_taxonomy_national_centers.
+      // eslint-disable-next-line max-len
+      const fieldTaxonomyNationalCenters = metadata.field_taxonomy_national_centers;

```

Comment:
![Screenshot 2023-06-08 at 3 23 25 PM](https://github.com/HHS/Head-Start-TTADP/assets/3288586/81ae13dc-87f0-4abb-b43b-e926eec60a8b)

this is the national center associated with the page, why do we need to also scrape the page? 

Since the title is in the JSON as well, why not simply (for ECLKC pages) try to get the metadata first, and if that fails, THEN try scraping the page?

Reply:
Are you saying get the national center name from the URL?

Reply:
From the acceptance criteria:

> if possible capture the following elements; ++  "created", "changed", "title", “field_taxonomy_national_centers", "field_taxonomy_topic", "langcode", "field_context" and associate with the URL, goal, Activity Report. 

I'm saying - I don't think there is any need to make 2 requests for an ECLKC page that has a JSON formatted response. The JSON response has everything we need. 

In the JSON response, we have a unique id (769 in my above screenshot) for that taxonomical term, and a unique url with a human readable name. Just save that to the database. Don't bother making a separate request and parsing the document at all in that case.

When or if we do utilize a National Centers table, we can create an eclkc_id column and make the relationship there.